### ISSUE TITLE
AWS.jl@2

### ISSUE THREAD

**Issue by mattBrzezinski** [JuliaCloud/AWS.jl#348]
There are a couple issues and feature requests which would be nice to introduce. I'm creating this issue to keep track of all breaking changes which the community would introduce for a v2 of this package.

- #224 : Always return `Vector{UInt8}` for `S3.get_object()`
- #341 : Support kwargs for optional parameters instead of passing in a Dictionary

After a list is created we can create a feature branch to merge changes into and then go to master.

**Comment by omus** [comment]
Note that #384 supersedes #224 by providing a more robust return type for AWS API calls.

**Comment by omus** [comment]
We may also want to consider switching to JSON3 for AWS@2 (https://github.com/JuliaCloud/AWS.jl/issues/438)

**Comment by mattBrzezinski** [comment]
> We may also want to consider switching to JSON3 for AWS@2 (#438)

Agreed. Also this is a known problem and one internally that we are wanting to address as well here at Invenia,

https://github.com/JuliaCloud/AWS.jl/issues/519

**Comment by omus** [comment]
An update on AWS.jl@v2. I've made some progress on this work and have been tracking the various issues for a v2 in the [2.0 milestone](https://github.com/JuliaCloud/AWS.jl/issues?q=is%3Aissue%20state%3Aopen%20milestone%3A2.0). There are a few outstanding tasks I'd love to tackle as part of a new major version since we need to make a breaking change but I'm definitely bandwidth constrained and may not have time to tackle this. If I do find that progress is stalled out I'll just release v2 as is.
